vault/request_handling.go

```diff
@@ -549,39 +545,26 @@ func (c *Core) handleLoginRequest(ctx context.Context, req *logical.Request) (re
 			return nil, nil, ErrInternalError
 		}
 
-		// Start off with the sys default value, and update according to period/TTL
-		// from resp.Auth
-		tokenTTL := sysView.DefaultLeaseTTL()
-
-		switch {
-		case auth.Period > time.Duration(0):
-			// Cap the period value to the sys max_ttl value. The auth backend should
-			// have checked for it on its login path, but we check here again for
-			// sanity.
-			if auth.Period > sysView.MaxLeaseTTL() {
-				auth.Period = sysView.MaxLeaseTTL()
-			}
-			tokenTTL = auth.Period
-		case auth.TTL > time.Duration(0):
-			// Cap the TTL value. The auth backend should have checked for it on its
-			// login path (e.g. a call to b.SanitizeTTL), but we check here again for
-			// sanity.
-			if auth.TTL > sysView.MaxLeaseTTL() {
-				auth.TTL = sysView.MaxLeaseTTL()
-			}
-			tokenTTL = auth.TTL
+		tokenTTL, warnings, err := calculateTTL(sysView, 0, auth.TTL, auth.Period, auth.MaxTTL, auth.ExplicitMaxTTL, time.Time{})
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, warning := range warnings {
+			resp.AddWarning(warning)
 		}
 
 		// Generate a token
 		te := TokenEntry{
-			Path:         req.Path,
-			Policies:     auth.Policies,
-			Meta:         auth.Metadata,
-			DisplayName:  auth.DisplayName,
-			CreationTime: time.Now().Unix(),
-			TTL:          tokenTTL,
-			NumUses:      auth.NumUses,
-			EntityID:     auth.EntityID,
+			Path:           req.Path,
+			Policies:       auth.Policies,
+			Meta:           auth.Metadata,
+			DisplayName:    auth.DisplayName,
+			CreationTime:   time.Now().Unix(),
+			TTL:            tokenTTL,
+			NumUses:        auth.NumUses,
+			EntityID:       auth.EntityID,
+			Period:         auth.Period,
```

Comment:
We don't encode a period or explicit max ttl currently because it's up to backends to decide what that value should be at each renewal. The reason is that if you adjust the period for e.g. an approle role you likely want to have it affect all tokens issued by that role, not just ones going forward. The fields in TokenEntry are currently only for when direct creation of a token is used. Do you see a good reason to change this?

Reply:
The is left over from how I interpreted how periods should work.  I originally thought that the period (or explicit max) should not change after the token is issues.  Over the iterations I've moved away from this and reverted the changes in the token store and backends.  This can safely be removed.